trl/trainer/grpo_trainer.py

```diff
@@ -450,6 +450,20 @@ def _move_model_to_vllm(self):
         ) as unwrapped_model:
             if is_compiled_module(unwrapped_model):
                 state_dict = unwrapped_model._orig_mod.state_dict()
+            elif isinstance(unwrapped_model, PeftModel):
+                unwrapped_model.merge_adapter()
+                state_dict = unwrapped_model.state_dict()
+                unwrapped_model.unmerge_adapter()
+                state_dict = {
```

Comment:
It could be helpful to add a comment here on how the state dict is changed and why.

Reply:
Maybe something like:

"Note that merge_adapter merges leaves the PEFT model in place, unlike merge_and_unload, but does not require a full copy. Therefore, we use this method, but the PEFT-specific prefixes need to be removed. Since the adapter weights are already merged into the base weights, we can remove those too. Next, adjust the prefix of modules_to_save (if present) and remove the corresponding original_module. After the state_dict creation, we can call unmerge_adapter to restore the PEFT model to its original state."

It's a bit verbose, not sure if that's an issue, but I prefer to give too much info rather than too little.